torch/ao/quantization/fx/prepare.py

```diff
@@ -908,7 +955,29 @@ def insert_observers_for_model(
     passes which optimize the graph by deduplicating observers, etc.
     """
 
-    node_name_to_target_dtype: Dict[str, Any] = {}
+    # name of Node in original FX Graph to the target dtype information
+    # that's derived from qconfig for the Node, for example, if we have
+    # a conv2d node that has a qconfig
+    # {
+    #   # information for input and bias node omitted
+    #   # for getattr node
+    #   # weight = getattr(self, 'weight')
+    #   'weight': {
+    #      'output_activation_dtype': torch.float,
+    #   }
+    #   # for conv2d node
+    #   # conv2d = call_function[target=torch.nn.functional.conv2d](
+    #   #            args=(input, weight, bias))
+    #   'conv2d': {
+    #       'input_activation_dtype': torch.quint8,
+    #       'weight_dtype': torch.qint8,
+    #       'bias_dtype': torch.float,
+    #       'output_activation_dtype': torch.quint8,
+    #     }
+    #   }
+    #
+    # TODO: rename this to node_name_to_target_dtype_info
+    node_name_to_target_dtype: Dict[str, Dict[str, Any]] = defaultdict(dict)
```

Comment:
should we rename this to something like `original_node_name_to_target_dtype_info`, to clarify that only nodes in the original graph are expected to be in this data structure?

Reply:
should the type be `Dict[str, Dict[str, Optional[torch.dtype]]]`?

Reply:
sounds good, will rename this in next PR.
will fix the type